app/waf/src/controllers/declaration.controller.ts

```diff
@@ -333,4 +342,69 @@ export class DeclarationController extends BaseController {
       .declarations(applicationId)
       .delete({and: [{id: declarationId}, {tenantId: await this.tenantId}]});
   }
+
+  @post(
+    prefix +
+      '/applications/{applicationId}/declarations/{declarationId}/deploy',
+    {
+      responses: {
+        '204': Schema.emptyResponse('Successfully deploy Declaration resource'),
+        '404': Schema.notFound('Can not find Declaration resource'),
+        '422': Schema.unprocessableEntity(
+          'Fail to deploy Declaration resource',
+        ),
+      },
+    },
+  )
+  async proxyDeploy(
+    @param(Schema.pathParameter('applicationId', 'Application resource ID'))
+    applicationId: string,
+    @param(Schema.pathParameter('declarationId', 'Declaration resource ID'))
+    declarationId: string,
+    @requestBody(
+      Schema.createRequest(
+        ASGDeployRequest,
+        'Deploy the declaration on an ADC.',
+      ),
+    )
+    deployBody: ASGDeployRequest,
+  ): Promise<void> {
+    let adcId = deployBody.adcId!;
+    await Promise.all([
+      this.applicationRepository.findById(applicationId, undefined, {
+        tenantId: await this.tenantId,
+      }),
+      this.declarationRepository.findById(declarationId, undefined, {
+        tenantId: await this.tenantId,
+      }),
+      this.adcRepository.findById(adcId, undefined, {
+        tenantId: await this.tenantId,
+      }),
+    ]).then(
+      async ([application, declaration, adc]) => {
+        // TODO: uncomment it; use adcstate control: gotTo(Active)
+        // if (adc.status !== 'ACTIVE' || !adc.management) {
+        //   throw new HttpErrors.UnprocessableEntity('ADC resource is not ready for deploy.');
+        // }
+
+        let mgmt = adc.management!;
+        let proxyMgr = await ASGManager.instanlize();
+
+        try {
+          let as3Request = new AS3DeployRequest(adc, application, declaration);
+          await proxyMgr.deploy(
+            mgmt.ipAddress,
+            mgmt.tcpPort,
+            as3Request.declaration,
```

Comment:
this is only passing as3Request.declaration. isn't the following also needed in the body?: e.g.   "class": "AS3",  "action": "deploy", etc.

Reply:
No need I think. So I'd prefer to remove all AS3 hints from the code. 

AS3DeployRequest may be useless, maybe...